### PR TITLE
Php js support

### PR DESCRIPTION
**Pull request by Neloop**

### CODE REVIEW

File: app/helpers/ExerciseConfig/Compilation/VariablesResolver.php [hunk 1]
```diff
@@ -120,8 +120,11 @@ public function resolveForInputNodes(MergeTree $mergeTree, VariablesTable $exerc
       $outputPort = current($inputBox->getOutputPorts());
       $variableName = $outputPort->getVariable();
       $child = current($node->getChildren());
-      $inputPortName = array_search($node, $child->getParents());
+      if ($child === false) {
+        throw new ExerciseConfigException("Input port not found for variable {$variableName}");
```
Comment (Teyras) [inline app/helpers/ExerciseConfig/Compilation/VariablesResolver.php]: Shouldn't this be handled by the validator?

File: app/helpers/ExerciseConfig/Compilation/PipelinesMerger.php [hunk 2]
```diff
@@ -142,7 +142,10 @@ private function mergeTrees(MergeTree $first, MergeTree $second): MergeTree {
     }
 
     // transform remaining output from first tree into classic array
-    $newFirstOutput = array_values($outVars);
+    $newFirstOutput = [];
+    foreach ($outVars as $outVar) {
```
Comment (Teyras) [inline app/helpers/ExerciseConfig/Compilation/PipelinesMerger.php]: If only `array_map` was not the verbose beast it is in PHP...

File: app/helpers/ExerciseConfig/Compilation/VariablesResolver.php (hunk 1 above)
Comment (Neloop) [reply app/helpers/ExerciseConfig/Compilation/VariablesResolver.php]: It definitely should, but was not... I ran into this error while testing PHP and better be sure that in there this cannot happen, I will create task for handling this in validator.

File: app/helpers/ExerciseConfig/Compilation/PipelinesMerger.php (hunk 2 above)
Comment (Neloop) [reply app/helpers/ExerciseConfig/Compilation/PipelinesMerger.php]: Well sorry, I am more like the classical foreach guy :-) but can change it if you want.

Comment (Teyras) [reply app/helpers/ExerciseConfig/Compilation/PipelinesMerger.php]: That was meant to be a criticism of your code, just a little rant. Foreach is OK.

File: app/helpers/ExerciseConfig/Compilation/VariablesResolver.php (hunk 1 above)
Comment (Teyras) [reply app/helpers/ExerciseConfig/Compilation/VariablesResolver.php]: OK, that sounds reasonable.

**Pull request merged**